Responsive in app surveys

Dhruwang opened the pull request:
## What does this PR do?

Makes in-app surveys mobile friendly
Adds functionality to preview survey in mobile device

Fixes #544 


https://github.com/formbricks/formbricks/assets/67850763/6edba914-72fb-405e-bc63-0972a93ba51b


https://github.com/formbricks/formbricks/assets/67850763/456c6022-443a-436a-88e7-371ad9df02e1



<!-- Please provide a screenshots or a loom video for visual changes to speed up reviews
 Loom Video: https://www.loom.com/
-->

## Type of change

<!-- Please mark the relevant points by using [x] -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] Enhancement (small improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->

Create an in-app survey and test on small devices
Go to preview survey page and toggle between mobile view and desktop view

## Checklist

<!-- We're starting to get more and more contributions. Please help us making this efficient for all of us and go through this checklist. Please tick off what you did  -->

- [x] Added a screen recording or screenshots to this PR
- [x] Filled out the "How to test" section in this PR
- [x] Read the [contributing guide](https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [x] Updated the Formbricks Docs if changes were necessary

Dhruwang commented on packages/js/src/components/Modal.tsx:
```diff
@@ -76,9 +99,12 @@ export default function Modal({
             getPlacementStyle(placement),
             show ? "fb-opacity-100" : "fb-opacity-0",
-            "fb-h-fit fb-pointer-events-auto fb-absolute fb-w-full fb-max-w-sm fb-overflow-hidden fb-rounded-lg fb-bg-white fb-shadow-lg fb-ring-1 fb-ring-black fb-ring-opacity-5 fb-transition-all fb-duration-500 fb-ease-in-out sm:fb-m-4"
+            "fb-h-fit fb-pointer-events-auto fb-absolute fb-w-full sm:fb-max-w-sm fb-overflow-hidden fb-rounded-lg fb-bg-white fb-shadow-lg fb-ring-1 fb-ring-black fb-ring-opacity-5 fb-transition-all fb-duration-500 fb-ease-in-out sm:fb-m-4",
```
Here fb-bottom-0 and fb-inset-x-0 makes the modal sticks to bottom on small devices

Dhruwang commented on packages/js/src/components/Modal.tsx:
```diff
@@ -76,9 +99,12 @@ export default function Modal({
             show ? "fb-opacity-100" : "fb-opacity-0",
-            "fb-h-fit fb-pointer-events-auto fb-absolute fb-w-full fb-max-w-sm fb-overflow-hidden fb-rounded-lg fb-bg-white fb-shadow-lg fb-ring-1 fb-ring-black fb-ring-opacity-5 fb-transition-all fb-duration-500 fb-ease-in-out sm:fb-m-4"
+            "fb-h-fit fb-pointer-events-auto fb-absolute fb-w-full sm:fb-max-w-sm fb-overflow-hidden fb-rounded-lg fb-bg-white fb-shadow-lg fb-ring-1 fb-ring-black fb-ring-opacity-5 fb-transition-all fb-duration-500 fb-ease-in-out sm:fb-m-4",
+            isMobile && "fb-top-full fb-rounded-t-3xl",
```
When on small screen size, modal stays below the screen initially and slides up when show value becomes true (next line)

jobenjada commented:
Hey @Dhruwang 

thanks a lot for this PR, really appreciate the full description and comments on the code - huge improvement! 🙏 ⚡ 

I've noticed a few things:

- The mobile preview always shows the in-app survey:
https://tella.video/johanness-video-2d8g

- The toggle is too big, especially on small screens. Broadly speaking, half the size should be sufficient:
![grafik](https://github.com/formbricks/formbricks/assets/72809645/a5570bb3-0cec-4272-a0f1-1a71d1dc0f88)

- The phone looks a bit clunky. Pls increase the height a bit (gray lines) and decrease the border width a bit (blue line):
![grafik](https://github.com/formbricks/formbricks/assets/72809645/a94c4cef-5415-4a3d-9da5-66a2b13340af)

Thanks for picking this one up - looks great! 😊

Dhruwang commented:
On it 😊

Dhruwang commented:
Changes:
Made mobile mockup border thinner
Increased mockup's height
Reduced toggle's size
Removed duplicates CSS classes
Fixed : The mobile preview always shows the in-app survey

https://github.com/formbricks/formbricks/assets/67850763/e53e6659-f92b-47cb-8195-db28c4075289

Dhruwang commented on apps/web/app/environments/[environmentId]/surveys/PreviewSurvey.tsx:
```diff
@@ -24,6 +26,68 @@ interface PreviewSurveyProps {
+  </div>
+);
+
+const QuestionRenderer = ({ activeQuestionId, lastActiveQuestionId, questions, brandColor, thankYouCard, gotoNextQuestion }) => {
```
Extracted question rendering logic into a separate component as it was being used multiple times

jobenjada commented:
Hey! 

We're getting there 🙏  I found a few more small tweaks / bugs:

1. Survey does not jump back to top in preview when two questions with scroll follow behind each other:
https://tella.video/johanness-video-3qzv

2. Reduce height of nodge a bit, there is not enough space between the question and the nodge
![grafik](https://github.com/formbricks/formbricks/assets/72809645/e2f2a216-dfd5-46d9-9b53-7d4b4e246196)

3. Max height of Link Survey container with too high to fully display toggle:
![grafik](https://github.com/formbricks/formbricks/assets/72809645/3debf257-41a9-4b08-8963-16d38422b59f)

The rest looks good! 🚀

Dhruwang commented:
On it 😊

Dhruwang commented:
Resolved above mentioned issues

https://github.com/formbricks/formbricks/assets/67850763/aef6e463-cef0-4720-8e12-c9591fb222b5


https://github.com/formbricks/formbricks/assets/67850763/69616c9b-4fed-453c-b482-6d5a759be204

jobenjada commented:
Hey @Dhruwang  !

Looks great, works smoothly 🚀  Can you please resolve the two merge conflicts?

@mattinannt from UI/UX ready to merge :)

Dhruwang commented:
Merge conflicts resolved
Also fixes a minor issue of back button disappearing from preview once survey is finished

https://github.com/formbricks/formbricks/assets/67850763/2555f0cc-d496-4298-a097-9649b94803c1

mattinannt commented:
@Dhruwang thanks for the PR 😊 It looks great in your video but when testing on my computer in the demo app, the popup is showing bottom left instead of bottom right and on mobile the popup isn't displayed at all (no errors in console, just not showing (css)). Can you please take a look? 😊

https://github.com/formbricks/formbricks/assets/675065/023a7cf7-8456-4501-8164-5f1ab7dda3b8

Dhruwang commented:
Have resolved the issue 😊 @mattinannt

mattinannt reviewed (approved):
@Dhruwang thanks a lot for making the changes 😊 The new preview looks really great and now also the web-app survey is working correctly 💪

I will merge the changes into main now 😊
